ledger/src/blockstore.rs

```diff
@@ -1532,6 +1532,26 @@ impl Blockstore {
             .collect())
     }
 
+    /// Returns a transaction status if it was processed in a root or pending slot
+    pub fn get_transaction_status(
+        &self,
+        signature: Signature,
+        maximum_slot: Slot,
+        root: Slot,
+    ) -> Result<Option<(Slot, RpcTransactionStatusMeta)>> {
+        let mut transaction_iter = self.transaction_status_cf.iter(IteratorMode::End)?;
```

Comment:
Main concern right now is iterating through the entire status column family for the ledger. I don't know enough about rocks to feel comfortable approving this.